numbert/utils/utils_numbert.py

```diff
@@ -316,6 +324,158 @@ def _create_examples(self, data, set_type):
+            doc_titles = [doc_titles for doc_titles, _ in doc_titles_ranks]
+            sorted_run[query_id] = doc_titles
+
+        return sorted_run
+
+    def load_collection(self, data_dir):
+        """Loads TREC-CAR's paraghaphs into a dict of key: title, value: paragraph."""
+        self.corpus = {}
+
+        start_time = time.time()
+        APPROX_TOTAL_PARAGRAPHS = 30000000
+        with open(os.path.join(data_dir, "paragraphCorpus/dedup.articles-paragraphs.cbor"), 'rb') as f:
+            for i, p in enumerate(trec_car_classes.iter_paragraphs(f)):
+                para_txt = [elem.text if isinstance(elem, trec_car_classes.ParaText)
+                            else elem.anchor_text for elem in p.bodies]
+                self.corpus[p.para_id] = ' '.join(para_txt)
+                if i % 10000 == 0:
+                    print('Loading paragraph {} of {}'.format(i, APPROX_TOTAL_PARAGRAPHS))
+                    time_passed = time.time() - start_time
+                    hours_remaining = (
+                        APPROX_TOTAL_PARAGRAPHS - i) * time_passed / (max(1.0, i) * 3600)
+                    print('Estimated hours remaining to load corpus: {}'.format(
+                        hours_remaining))
+
+    def merge(self, qrels, run):
+        """Merge qrels and runs into a single dict of key: query_id, 
+        value: tuple(relevant_doc_ids, candidate_doc_ids)"""
+        data = collections.OrderedDict()
+        for query_id, candidate_doc_ids in run.items():
+            relevant_doc_ids = set()
+            if qrels:
+                relevant_doc_ids = qrels[query_id]
+            data[query_id] = (relevant_doc_ids, candidate_doc_ids)
+        return data
+
+    def get_train_examples(self, data_dir, is_qrels = True):
+        """See base class."""
+        qrels = None
+        if is_qrels:
+            qrels = self.load_qrels(os.path.join(data_dir,"train.qrels"))
+        
+        run = self.load_run(os.path.join(data_dir,"train.run"))
+        eval_data = self.merge(qrels=qrels, run=run)
+
+        return self._create_examples(eval_data, "train")
+
+    def get_dev_examples(self, data_dir, is_qrels = True):
+        """See base class."""
+        qrels = None
+        if is_qrels:
+            qrels = self.load_qrels(os.path.join(data_dir,"dev.qrels"))
+
+        run = self.load_run(os.path.join(data_dir,"dev.run"))
+        dev_data = self.merge(qrels=qrels, run=run)
+
+        return self._create_examples(dev_data, "dev")
+
+    def get_test_examples(self, data_dir, is_qrels = True):
+        """See base class."""
+        qrels = None
+        if is_qrels:
+            qrels = self.load_qrels(os.path.join(data_dir,"test.qrels"))
+
+        run = self.load_run(os.path.join(data_dir,"test.run"))
+        eval_data = self.merge(qrels=qrels, run=run)
+
+        return self._create_examples(eval_data, "eval")
+
+
+    def get_examples_online(self, queries, data):
+        """Creates examples for online setting."""
```

Comment:
Why is it called online? Maybe add more details? Or did I miss its explanation somewhere?

Reply:
Online means interactive here, happening in an online fashion,  Will adding "online interactive setting" does that clarify it?

Reply:
Added to the comment in newest push

Reply:
Yes, much better, thanks!